test/tools/spec-runner/context.js

```diff
@@ -87,11 +87,13 @@ class TestRunnerContext {
     return this.runForAllClients(client => client.connect());
   }
 
-  teardown() {
-    return Promise.all([
-      this.runForAllClients(client => client.close()),
-      this.sharedClient.close()
-    ]);
+  async teardown() {
+    await this.runForAllClients(async client => {
+      if (client) await client.close();
+    });
+    if (this.sharedClient) {
```

Comment:
just a note that with the Promise.all, the first thing failing doesn't prevent the second from executing, but with this setup, if the runForAllClients fails, this sharedClient block won't execute

Reply:
We should be able to change this back with the new way the config file works. This was happening where all the int tests are skipped because there's no mongodb running but some of the after hooks didn't gate their calls the way before checks did

Reply:
Ok actually I simplified it, reading runForAllClients we were calling close on the sharedClient twice